[263] Challenge manager dashboard updates

stonefilipczak opened the pull request:
Responds to testing feedback on ticket 263: 
<img width="638" alt="Screen Shot 2025-03-06 at 2 54 30 PM" src="https://github.com/user-attachments/assets/fb00a91b-71f8-451f-a62e-3240ceadcc0f" />

- H1s are made normal thickness app-wide
- Table styling for our custom "gray-header" uswds table is updated to include a blue top border app-wide
- link color is updated app-wide
- tag color was addressed in a previous PR
- all field values are sentence case
- evaluation due date field values are formatted consistently with other dates on the page
- page width was addressed in a previous PR
<img width="1120" alt="Screen Shot 2025-03-06 at 2 58 01 PM" src="https://github.com/user-attachments/assets/8dccfc26-bdde-4387-8d10-106a9b965454" />

stonefilipczak committed:
un-bold h1s app-wide

stonefilipczak committed:
add blue top border to our custom tables

stonefilipczak committed:
change link colors app wide

stonefilipczak committed:
sentence case cell values

stonefilipczak committed:
format eval due date

stonefilipczak committed:
fix tests

stepchud committed:
Merge branch 'dev' into challenge_manager_dashboard_updates

stepchud committed:
Merge branch 'dev' into challenge_manager_dashboard_updates

stepchud commented on app/assets/stylesheets/application.sass.scss:
```diff
@@ -41,6 +41,12 @@ dialog::backdrop {
+
+  tbody {
+    tr {
+      border-top: 4px solid #005EA2;
```
This should use a global theme color constant instead of hard-coding.

stepchud replied:
Also it doesn't match the design
figma:
<img width="478" alt="Screenshot 2025-03-07 at 1 50 31 PM" src="https://github.com/user-attachments/assets/53587173-2cc1-460e-ae1e-54b47b45a082" />
pr:
![Screenshot 2025-03-07 at 1 51 11 PM](https://github.com/user-attachments/assets/0f461842-8030-48b1-8732-cac3be2ef95c)

stepchud commented on app/assets/uswds/_uswds-theme.scss:
```diff
@@ -73,3 +73,11 @@ Add a list of changed settings in the form $setting: value.
+
+a {
+  color: #005EA2;
+}
```
1. The links on figma have a few different states with colors defined
    <img width="741" alt="Screenshot 2025-03-07 at 12 56 53 PM" src="https://github.com/user-attachments/assets/ff64f70b-46e0-4bc0-9842-87d53583fea0" />
2. imo global colors should modify the uswds themes instead of applying a global CSS rule to all links

stepchud committed:
customize styles

- table.challenge-table has special styling for borders
- move layout sass to uswds to use theme-colors
- contained flash messages

stepchud committed:
table style

stepchud reviewed (approved):
looks good to me

stepchud committed:
codeclimate
